src/ralph/backends/lrs/mongo.py

```diff
@@ -0,0 +1,90 @@
+
+import logging
+from typing import Iterator, List
+
+from bson.objectid import ObjectId
+from pymongo import ASCENDING, DESCENDING
+
+from ralph.backends.data.mongo import MongoDataBackend
+from ralph.backends.lrs.base import (
+    BaseLRSBackend,
+    StatementParameters,
+    StatementQueryResult,
+)
+from ralph.exceptions import BackendException, BackendParameterException
+
+logger = logging.getLogger(__name__)
+
+
+class MongoLRSBackend(BaseLRSBackend, MongoDataBackend):
+    """MongoDB LRS backend."""
+
+    def query_statements(self, params: StatementParameters) -> StatementQueryResult:
+        """Return the results of a statements query using xAPI parameters."""
+        mongo_query_filters = {}
+
+        if params.statementId:
+            mongo_query_filters.update({"_source.id": params.statementId})
+
+        if params.agent:
+            mongo_query_filters.update({"_source.actor.account.name": params.agent})
+
+        if params.verb:
+            mongo_query_filters.update({"_source.verb.id": params.verb})
+
+        if params.activity:
+            mongo_query_filters.update(
+                {
+                    "_source.object.objectType": "Activity",
+                    "_source.object.id": params.activity,
+                },
+            )
+
+        if params.since:
+            mongo_query_filters.update(
+                {"_source.timestamp": {"$gt": params.since.isoformat()}}
+            )
+
+        if params.until:
+            mongo_query_filters.update(
+                {"_source.timestamp": {"$lte": params.until.isoformat()}}
+            )
+
+        if params.search_after:
+            search_order = "$gt" if params.ascending else "$lt"
+            mongo_query_filters.update(
+                {"_id": {search_order: ObjectId(params.search_after)}}
+            )
+
+        mongo_sort_order = ASCENDING if params.ascending else DESCENDING
+        mongo_query_sort = [
+            ("_source.timestamp", mongo_sort_order),
+            ("_id", mongo_sort_order),
+        ]
+
+        query = {
+            "filter": mongo_query_filters,
+            "limit": params.limit,
+            "sort": mongo_query_sort,
+        }
+        try:
+            mongo_response = list(self.read(query=query, ignore_errors=True))
+        except (BackendException, BackendParameterException) as error:
+            logger.error("Failed to read from MongoDB")
+            raise error
+
+        search_after = None
+        if mongo_response:
+            search_after = mongo_response[-1]["_id"]
+
+        return StatementQueryResult(
+            statements=[document["_source"] for document in mongo_response],
+            pit_id=None,
+            search_after=search_after,
+        )
+
+    def query_statements_by_ids(self, ids: List[str]) -> Iterator[dict]:
+        """Yield statements with matching ids from the backend."""
+        mongo_response = self.read(query={"filter": {"_source.id": {"$in": ids}}})
```

Comment:
```suggestion
        try:
            mongo_response = self.read(query={"filter": {"_source.id": {"$in": ids}}})
        except (BackendException, BackendParameterException) as error:
            logger.error("Failed to read from MongoDB")
            raise error
```

Reply:
Well spotted) Thanks) Updated this part)